numpy/random/_common.pyx

```diff
@@ -408,9 +408,18 @@ cdef int check_array_constraint(np.ndarray val, object name, constraint_type con
             raise ValueError("{0} value too large".format(name))
         elif not np.all(np.greater_equal(val, 0.0)):
             raise ValueError("{0} < 0 or {0} contains NaNs".format(name))
+    elif cons == CONS_NEGATIVE_BINOMIAL_POISSON:
```

Comment:
Isn't this redundant with `CONS_POISSON`.  Seems like the only difference is possibly in the exception.